File: src/main/java/it/contrader/model/Device.java
package it.contrader.model;

import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 
 * Model dell'entità User. Contiene l'enum che definisce gli usertype (salvati come interi
 * a partire da 0 sul db).
 * 
 * @author Vittorio Valent & Girolamo Murdaca
 * 
 * @see UserDTO
 */
@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Device {
	

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long id;
	
	@ManyToOne
	private Client client;
	
	private String mac;
	
	private String devtype;
	
	private String position;
	
}

	

File: src/dto/device.ts
import { Client } from './client'

export class Device {

    idDevice : number

    client : Client

    mac :string

    devtype : string

    position : string

    constructor(idDevice : number, client : Client, mac :string, devtype : string, position : string){

        this.idDevice = idDevice
        this.client = client
        this.mac = mac
        this.devtype = devtype
        this.position = position
    }
}